nautobot_golden_config/forms.py

```diff
@@ -389,3 +400,125 @@ class Meta:
+        label="Feature",
+        to_field_name="name",
+    )
+    change_control_id = forms.CharField(required=False, label="Change Control ID")
+    status = utilities_forms.DynamicModelMultipleChoiceField(
+        required=False,
+        queryset=Status.objects.all(),
+        query_params={"content_types": models.ConfigPlan._meta.label_lower},
+        display_field="label",
+        label="Status",
+        to_field_name="name",
+    )
+    tag = TagFilterField(model)
+
+
+class ConfigPlanCreateForm(BootstrapMixin, forms.Form):
+    """Create Form for ConfigPlan instances."""
+
+    # Common create fields
+    change_control_id = forms.CharField(required=False, label="Change Control ID")
+
+    # Common filter fields
+    tenant_group = utilities_forms.DynamicModelMultipleChoiceField(queryset=TenantGroup.objects.all(), required=False)
+    tenant = utilities_forms.DynamicModelMultipleChoiceField(queryset=Tenant.objects.all(), required=False)
+    location = utilities_forms.DynamicModelMultipleChoiceField(queryset=Location.objects.all(), required=False)
+    region = utilities_forms.DynamicModelMultipleChoiceField(queryset=Region.objects.all(), required=False)
+    site = utilities_forms.DynamicModelMultipleChoiceField(queryset=Site.objects.all(), required=False)
+    rack_group = utilities_forms.DynamicModelMultipleChoiceField(queryset=RackGroup.objects.all(), required=False)
+    rack = utilities_forms.DynamicModelMultipleChoiceField(queryset=Rack.objects.all(), required=False)
+    role = utilities_forms.DynamicModelMultipleChoiceField(queryset=DeviceRole.objects.all(), required=False)
+    manufacturer = utilities_forms.DynamicModelMultipleChoiceField(queryset=Manufacturer.objects.all(), required=False)
+    platform = utilities_forms.DynamicModelMultipleChoiceField(queryset=Platform.objects.all(), required=False)
+    device_type = utilities_forms.DynamicModelMultipleChoiceField(queryset=DeviceType.objects.all(), required=False)
+    device = utilities_forms.DynamicModelMultipleChoiceField(queryset=Device.objects.all(), required=False)
+    tag = utilities_forms.DynamicModelMultipleChoiceField(
+        queryset=Tag.objects.all(), query_params={"content_types": "dcim.device"}, required=False
+    )
+    status = utilities_forms.DynamicModelMultipleChoiceField(
+        queryset=Status.objects.all(), query_params={"content_types": "dcim.device"}, required=False
+    )
+
+
+class ConfigPlanCreateFeatureForm(ConfigPlanCreateForm):
+    """Create Form for ConfigPlan with features instances."""
+
+    features = utilities_forms.DynamicModelMultipleChoiceField(
+        queryset=models.ComplianceFeature.objects.all(),
+        display_field="name",
+        required=False,
+        help_text="Note: Selecting no features will generate plans for all applicable features.",
+    )
+
+
+class ConfigPlanCreateCommandsForm(ConfigPlanCreateForm):
+    """Create Form for ConfigPlan with commands instances."""
+
+    commands = forms.CharField(
```

Comment:
just wondering if we should follow a similar pattern as core does with secrets parameter which allows for jinja2 interpoliation.
https://github.com/nautobot/nautobot/blob/ad9fd96170f696b56c38ae3c20d51d5656812e4b/nautobot/extras/forms/forms.py#L1271C13-L1271C13
Just as far as using JSONField instead.

Reply:
looks like we can disregard this one.